### PR TITLE
Dockerfile: appcontainer for every arch

### LINKED ISSUE #93: `docker build` fails on Mac M1
Trying to build the docker image with:
```sh
docker build -t planutils:latest .
```
I get the following error:
```
#6 2.523  27000K .......... .......... .......... .......... .......... 99% 14.1M 0s
#6 2.526  27050K ......                                                100%  276M=1.7s
#6 2.526 
#6 2.526 2022-07-05 15:15:43 (15.8 MB/s) - 'apptainer_1.0.2_amd64.deb' saved [27705364/27705364]
#6 2.526 
#6 2.538 dpkg: error processing archive apptainer_1.0.2_amd64.deb (--install):
#6 2.538  package architecture (amd64) does not match system (arm64)
#6 2.542 Errors were encountered while processing:
#6 2.542  apptainer_1.0.2_amd64.deb
------
executor failed running [/bin/sh -c wget https://github.com/apptainer/apptainer/releases/download/v1.0.2/apptainer_1.0.2_amd64.deb     && dpkg -i apptainer_1.0.2_amd64.deb     && rm apptainer_1.0.2_amd64.deb]: exit code: 1
Arrêt du processus de terminal "/bin/zsh '-c', 'docker build --pull --rm -f "Dockerfile" -t planutils:latest "."'". Code de sortie : 1.
```
I am working on a Mac M1, hence it works with arm64 instead of amd64.

### PR DESCRIPTION
**Pull request by victorpaleologue**
Using a script found online,
it is easy to recompile apptainer
to ensure compatibility with the docker's architecture.

May fix #93

### CODE REVIEW

**Comment by FlorianPommerening** [comment]
I would suggest two changes. Both of them assume that the installation through the deb package is more stable than the installation from source and will more likely work with future releases. I see the installation from source a bit as a workaround/fallback since there is no prepackaged arm release, but I could imagine that they'll add one in the future (once more mac users complain). So with this in mind, I would:
1) move the installation of additional packages into the script (they will not be needed anymore if there is an arm release).
2) for x64, leave the installation through the deb package.

This would mean the Dockerfile would do something like (the bash stuff is from memory, so might not work)

```Dockerfile
RUN if [ $(arch) == "x86_64" ]; \
    then \
        wget https://github.com/apptainer/apptainer/releases/download/v1.0.2/apptainer_1.0.2_amd64.deb \
        && dpkg -i apptainer_1.0.2_amd64.deb \
        && rm apptainer_1.0.2_amd64.deb ; \
    else; \
        ./install-apptainer-from-source.sh
    fi
```

**Comment by victorpaleologue** [comment]
> This is for arch. Is this compatible with the ubuntu base image we are using? If so, it is probably preferable to an installation from source.

I do not agree. If it works, it would be by chance, and there is no good reason it keeps working. Whereas by building from source, you make sure it is compatible with the rest of the system.

**Comment by FlorianPommerening** [comment]
Sorry for taking so long to react here. I think the changes in https://github.com/AI-Planning/planutils/commit/0d5a1d1af200efb64937f7efd1b37dd7cc972787 look good. I have two small comments:

First, if the script installs stuff through `apt`, and does `apt update`, it will leave some stuff in `/var/lib/apt/lists`. As I understand it, Dockerfiles usually clean this up to avoid large images. Even though it is cleaned up later in the Dockerfile, this happens in a different layer, which means it will increase the size of the final container. A simple solution would be to move the execution of the script from a separate run command into the `apt-get update && apt apt-get install ... && rm -rf /var/lib/apt/lists/*`  block. I think it also fits there because it is also installing a dependency.

Second, the script takes the deb from https://github.com/apptainer/apptainer but the sources from https://github.com/hpcng/singularity. Is there a reason not to take Apptainer for both? In particular, the hpcng repo has this note, in their "about" section: "Singularity has been EOL'ed, see Apptainer".

**Comment by victorpaleologue** [comment]
> A simple solution would be to move the execution of the script from a separate run command into the `apt-get update && apt apt-get install ... && rm -rf /var/lib/apt/lists/*` block. I think it also fits there because it is also installing a dependency.

Ok why not. Then the setup of `pip` and `planutils` using several `RUN` commands also deserve to be regrouped. I won't do it here though.

> Second, the script takes the deb from https://github.com/apptainer/apptainer but the sources from https://github.com/hpcng/singularity. Is there a reason not to take Apptainer for both? In particular, the hpcng repo has this note, in their "about" section: "Singularity has been EOL'ed, see Apptainer".

Woopsie my bad. I imported that script from a gist, I did not realize it was obsolete.

**Comment by victorpaleologue** [comment]
Hello, please note that the PR is ready for review.
There is no hurry, though.
I know the start of school year is challenging.

**Comment by FlorianPommerening** [comment]
Thanks for the notification. I missed the one from Github.

I have one more quibble:
The package installation always installs 1.0.2:
```
wget https://github.com/apptainer/apptainer/releases/download/v1.0.2/apptainer_1.0.2_amd64.deb
```
while the installation from source takes its version from the script parameters:
```
APPTAINER_BASE_NAME=apptainer-$3
wget https://github.com/apptainer/apptainer/releases/download/v$3/$APPTAINER_BASE_NAME.tar.gz
```

It would make sense to unify this, so calling the script has a more consistent effect (either always installing 1.0.2 or always installing a version given as the parameter). In that direction, I would also change the order of the parameters to mention the apptainer version first (this is the main parameter of the script) and maybe hard-code "linux" as the target OS (the scriptname "install-apptainer-ubuntu.sh" already suggests that the target platform is Ubuntu and since we use apt inside, it is not clear what other parameter choices would make sense).

One more thing: After `make install`, can we safely clean up the directory containing the source code and the tar file? This would reduce the resulting image size.

**Comment by victorpaleologue** [comment]
There you go =)

File: install-apptainer-ubuntu.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+# exit when any command fails
+set -e
+
+# Install apptainer from release package on amd64
+ARCH=$(arch)
+if [ $ARCH == "x86_64" ]; then
+  DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y wget
+  DEBIAN_PACKAGE_NAME=apptainer_$1_amd64.deb
+  wget https://github.com/apptainer/apptainer/releases/download/v$3/$DEBIAN_PACKAGE_NAME
```
Comment (FlorianPommerening) [inline install-apptainer-ubuntu.sh:11]: `$3` -> `$1` (maybe easier to define variables at the top of the script, so it is clear what each parameter means)

Comment (victorpaleologue) [reply install-apptainer-ubuntu.sh]: Done

**Comment by FlorianPommerening** [comment]
I tried this locally and the build failed for me. So I fixed some things in the script (see details in the commit message) and pushed directly to your PR, I hope this is fine. The build on my PC works now. If it also works for you, I think we can merge this. Note that I removed `curl` because I think it is not used, so please test again to make sure.

**Comment by victorpaleologue** [comment]
It runs fine here, thanks for your test and fixes.

**Comment by FlorianPommerening** [comment]
Merged. Thanks for the contribution!

**Pull request merged**